sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs

```diff
@@ -116,8 +115,10 @@ internal ArmResource(ClientContext clientContext, ResourceIdentifier id)
         /// <param name="identifier"> The resource identifier. </param>
         protected virtual void ValidateResourceType(ResourceIdentifier identifier)
         {
-            if (identifier?.ResourceType != ValidResourceType)
+#if DEBUG
```

Comment:
why are we skipping only the default implementation of this method but not the overridden one?

Reply:
thats part of another PR where we are moving this out of the base class and into the static implementation we discussed